bba/bba_internal_test.go

```diff
@@ -0,0 +1,435 @@
+			}
+		}
+	})
+	tester.setupAssert(9, func(t *testing.T, bbaInstance *BBA) {
+		if bbaInstance.binValueSet.exist(one) {
+			t.Fatalf("binValue set **include** f+1 sent binary value")
+		}
+		if bbaInstance.binValueSet.exist(zero) {
+			t.Fatalf("binValue set **include** f+1 sent binary value")
+		}
+	})
+
+	for i, bval := range bvalList {
+		sender := cleisthenes.Member{
+			Address: cleisthenes.Address{
+				Ip:   "localhost" + strconv.Itoa(i),
+				Port: 8080,
+			},
+		}
+		if err := bbaInstance.handleBvalRequest(sender, bval); err != nil {
+			t.Fatalf("handle bval request failed with error: %s", err.Error())
+		}
+
+		if f, ok := tester.assert(i); ok {
+			f(t, bbaInstance)
+		}
+	}
+	done <- struct{}{}
+}
+
+func handleAuxRequestTestSetup(t *testing.T, auxList []*auxRequest) (*BBA, *mock.Broadcaster, *bbaTester, func()) {
+	tester := newBbaTester()
+	reqList, err := tester.setupAuxRequestList(auxList)
+	if err != nil {
+		t.Fatalf("failed to setup request list: err=%s", err)
+	}
+	bvalRepo := tester.setupRequestRepository()
+	auxRepo := tester.setupRequestRepository()
+	broadcaster := tester.setupBroadcaster(reqList)
+
+	bbaInstance := &BBA{
+		n:                   10,
+		f:                   3,
+		bvalRepo:            bvalRepo,
+		auxRepo:             auxRepo,
+		tryoutAgreementChan: make(chan struct{}, 1),
+		binValueSet:         newBinarySet(),
+		broadcastedBvalSet:  newBinarySet(),
+		broadcaster:         broadcaster,
+	}
+	return bbaInstance, broadcaster, tester, func() {
+		close(bbaInstance.tryoutAgreementChan)
+	}
+}
+
+func TestBBA_HandleAuxRequest(t *testing.T) {
+	auxList := []*auxRequest{
+		{Value: one},
+		{Value: one},
+		{Value: one},
+		{Value: one},
+		{Value: one},
+		{Value: one},
+		{Value: one},
+		{Value: one},
+	}
+
+	bbaInstance, _, tester, teardown := handleAuxRequestTestSetup(t, auxList)
+	defer teardown()
+
+	tester.setupAssert(6, func(t *testing.T, bbaInstance *BBA) {
+		// should receive signal from tryoutAgreementChan
+		<-bbaInstance.tryoutAgreementChan
+
+		reqList := bbaInstance.auxRepo.FindAll()
+		if len(reqList) != 7 {
+			t.Fatalf("expected request list length is %d, but got %d", 7, len(reqList))
+		}
+		for i, req := range reqList {
+			aux, ok := req.(*auxRequest)
+			if !ok {
+				t.Fatalf("request is not auxRequest type: [%d]", i)
+			}
+			if aux.Value != one {
+				t.Fatalf("request value is not one")
+			}
+		}
+	})
+
+	for i, aux := range auxList {
+		sender := cleisthenes.Member{
+			Address: cleisthenes.Address{
+				Ip:   "localhost" + strconv.Itoa(i),
+				Port: 8080,
+			},
+		}
+		if err := bbaInstance.handleAuxRequest(sender, aux); err != nil {
+			t.Fatalf("handle bval request failed with error: %s", err.Error())
+		}
+
+		if f, ok := tester.assert(i); ok {
+			f(t, bbaInstance)
+		}
+	}
+}
+
+func TestBBA_HandleAuxRequest_OneZeroCombined(t *testing.T) {
+	auxList := []*auxRequest{
+		{Value: one},  // 0. (one, zero) = (1, 0)
```

Comment:
what is scenario? 
1. 11 nodes? 
2. I know as `binVal` can has only one values.